### PR TITLE
changing stored timestamp from seconds to milliseconds

### PR DESCRIPTION
**Pull request by zhlu9890**
- updated lib/SampleService/core/storage/arango_sample_storage.py to store timestamp in milliseconds integer
- updated test/core/storage/arango_sample_storage_test.py to accommodate the code change
- added test/core/storage/arango_sample_storage_scrubber_test.py to test scrubber script for updating timestamp in existing records

```
        FOR sample1 IN samples_nodes
            FILTER sample1.saved < 1000000000000
            UPDATE sample1 WITH { saved: ROUND(sample1.saved * 1000) } IN samples_nodes
        FOR sample2 IN samples_version
            FILTER sample2.saved < 1000000000000
            UPDATE sample2 WITH { saved: ROUND(sample2.saved * 1000) } IN samples_version
        FOR link IN samples_data_link
            FILTER link.expired < 1000000000000 OR link.created < 1000000000000
            UPDATE link WITH {
                expired: link.expired < 1000000000000 ? ROUND(link.expired * 1000) : link.expired,
                created: link.created < 1000000000000 ? ROUND(link.created * 1000) : link.created
            } IN samples_data_link
```

### CODE REVIEW

File: test/core/storage/arango_sample_storage_scrubber_test.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    assert samplestorage.get_sample(id2).savetime == dt(ts3)
+    assert samplestorage.get_data_link(lid1).created == dt(ts2)
+    assert samplestorage.get_data_link(lid2).created == dt(ts3)
+    assert samplestorage.get_data_link(lid3).created == dt(ts3)
+    assert samplestorage.get_data_link(lid3).expired == dt(ts3+100)
+
+    samplestorage._db.aql.execute(
+        """
+        FOR sample1 IN samples_nodes
+            FILTER sample1.saved < 1000000000000
```
Comment (scanon) [inline test/core/storage/arango_sample_storage_scrubber_test.py:161]: Let's put this make this lower threshold a parameter and maybe explain the rationale.

File: test/core/storage/arango_sample_storage_scrubber_test.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    )
+
+    assert samplestorage.get_sample(id1, 1).savetime == dt(ts2)
+    assert samplestorage.get_sample(id1, 2).savetime == dt(ts2)
+    assert samplestorage.get_sample(id2).savetime == dt(ts2)
+    assert samplestorage.get_data_link(lid1).created == dt(ts2)
+    assert samplestorage.get_data_link(lid2).created == dt(ts2)
+    assert samplestorage.get_data_link(lid3).created == dt(ts2)
+    assert samplestorage.get_data_link(lid3).expired == dt((ts3+100) * 1000)
+
```
Comment (scanon) [inline test/core/storage/arango_sample_storage_scrubber_test.py:182]: Let's rerun that same update and confirm that it doesn't re-update things (e.g. multiply by another 1000).

**Review by scanon — CHANGES_REQUESTED** [review]
I think this looks okay but I would like another set of eyes on it.

**Comment by slebras** [comment]
looks good to me, i don't totally understand the Arango query completely in the tests.

File: test/core/storage/arango_sample_storage_scrubber_test.py (hunk 2 above)
Comment (zhlu9890) [reply test/core/storage/arango_sample_storage_scrubber_test.py:211]: I added another set of tests to confirm it doesn't re-update if timestamp is above the threshold

**Review by scanon — APPROVED** [review]
Looks good

**Pull request merged**